keylime-agent/src/main.rs

```diff
@@ -278,6 +278,56 @@ async fn main() -> Result<()> {
         config.agent.tpm_signing_alg.as_ref(),
     )?;
 
+    let (asym_alg, name_alg) = match config.agent.iak_idevid_template.as_str() {
+        "H-1" => {
+            (tss_esapi::interface_types::algorithm::AsymmetricAlgorithm::Rsa,
+            tss_esapi::interface_types::algorithm::HashingAlgorithm::Sha256)
+        },
+        "H-2" => {
+            (tss_esapi::interface_types::algorithm::AsymmetricAlgorithm::Ecc,
+            tss_esapi::interface_types::algorithm::HashingAlgorithm::Sha256)
+        },
+        "H-3" => {
+            (tss_esapi::interface_types::algorithm::AsymmetricAlgorithm::Ecc,
+            tss_esapi::interface_types::algorithm::HashingAlgorithm::Sha384)
+        },
+        "H-4" => {
+            (tss_esapi::interface_types::algorithm::AsymmetricAlgorithm::Ecc,
+            tss_esapi::interface_types::algorithm::HashingAlgorithm::Sha512)
+        },
+        "H-5" => {
+            (tss_esapi::interface_types::algorithm::AsymmetricAlgorithm::Ecc,
+            tss_esapi::interface_types::algorithm::HashingAlgorithm::Sm3_256)
+        },
+        _ => {
+            (match config.agent.iak_idevid_asymmetric_alg.as_str() {
+                "rsa" => {tss_esapi::interface_types::algorithm::AsymmetricAlgorithm::Rsa},
+                "ecc" => {tss_esapi::interface_types::algorithm::AsymmetricAlgorithm::Ecc},
+                _ => {tss_esapi::interface_types::algorithm::AsymmetricAlgorithm::Null},
+            },
+            match config.agent.iak_idevid_name_alg.as_str() {
+                "sha256" => {tss_esapi::interface_types::algorithm::HashingAlgorithm::Sha256},
+                "sm3_256" => {tss_esapi::interface_types::algorithm::HashingAlgorithm::Sm3_256},
+                "sha384" => {tss_esapi::interface_types::algorithm::HashingAlgorithm::Sha384},
+                "sha512" => {tss_esapi::interface_types::algorithm::HashingAlgorithm::Sha512},
+                _ => {tss_esapi::interface_types::algorithm::HashingAlgorithm::Null},
+            })
+        },
+    };
+
+    let (iak, idevid) = if config.agent.enable_iak_idevid {
+        let idevid = ctx.create_idevid(asym_alg, name_alg)?;
+        info!("IDevID created.");
+        // Flush after creating to make room for AK and EK and IAK
+        ctx.as_mut().flush_context(idevid.handle.into())?;
+
+        let iak = ctx.create_iak(asym_alg, name_alg)?;
+        info!("IAK created.");
+        (Some(iak), Some(idevid))
+    } else {
+        (None, None)
```

Comment:
Again, wouldn't be better to move this to a dedicated lib?

The API could return the Tuple `(iak, idevid)` or `(None, None)`

Reply:
This I think makes sense here along with the similar `create_ak` and `create_ek` functions

Reply:
Ok, we can leave this in `main.rs`